src/protocol_key.c

```diff
@@ -402,14 +403,14 @@ bool ans_key_h(connection_t *c, const char *request) {
 	char key[MAX_STRING_SIZE];
 	char address[MAX_STRING_SIZE] = "";
 	char port[MAX_STRING_SIZE] = "";
-	int cipher, digest;
+	nid_t cipher, digest;
 	unsigned long maclength;
-	int compression;
+	compression_level_t compression;
 	node_t *from, *to;
 
 	if(sscanf(request, "%*d "MAX_STRING" "MAX_STRING" "MAX_STRING" %d %d %lu %d "MAX_STRING" "MAX_STRING,
 	                from_name, to_name, key, &cipher, &digest, &maclength,
-	                &compression, address, port) < 7) {
+	                (int *)&compression, address, port) < 7) {
```

Comment:
I've been bitten by code like this in the past. It's safer to keep the local variables `int`, read into those, and then cast it to `nid_t`/`compression_level_t` later.

Reply:
Right, thank you for having a look at it.

I think `compression_level` in `connection_t` isn't actually used anymore, and its comparison with debug level here felt a bit weird:

https://github.com/gsliepen/tinc/blob/1.1/src/logger.c#L113-L115

so I split it into two fields and then removed the old one.